src/mlpack/methods/decision_tree/random_binary_numeric_split_impl.hpp

```diff
@@ -0,0 +1,151 @@
+/**
+ * @file methods/decision_tree/random_binary_numeric_split_impl.hpp
+ * @author Rishabh Garg
+ *
+ * Implementation of strategy that finds the random binary numeric split.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_DECISION_TREE_RANDOM_BINARY_NUMERIC_SPLIT_IMPL_HPP
+#define MLPACK_METHODS_DECISION_TREE_RANDOM_BINARY_NUMERIC_SPLIT_IMPL_HPP
+
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace tree {
+
+template<typename FitnessFunction>
+template<bool UseWeights, typename VecType, typename WeightVecType>
+double RandomBinaryNumericSplit<FitnessFunction>::SplitIfBetter(
+    const double bestGain,
+    const VecType& data,
+    const arma::Row<size_t>& labels,
+    const size_t numClasses,
+    const WeightVecType& weights,
+    const size_t minimumLeafSize,
+    const double minimumGainSplit,
+    arma::vec& classProbabilities,
+    AuxiliarySplitInfo& /* aux */)
+{
+  double bestFoundGain = std::min(bestGain + minimumGainSplit, 0.0);
+  // Forcing a minimum leaf size of 1 (empty children don't make sense).
+  const size_t minimum = std::max(minimumLeafSize, (size_t) 1);
+
+  // First sanity check: if we don't have enough points, we can't split.
+  if (data.n_elem < (minimum * 2))
```

Comment:
Why the `* 2` here?

Reply:
Since it is a binary split, so if the number of elements is less than `minimum * 2` then there will be at least one child which will have the number of elements less than `minimumLeafSize`. And we want every leaf to have at least `minimumLeafSize` elements. That's why `* 2` is there. Also, this check is in the `BestBinaryNumericSplit` you can also check there :)